monai/apps/pathology/transforms/spatial/array.py

```diff
@@ -73,3 +75,153 @@ def get_params(self, image_size):
+        else:
+            self.random_idxs = None
+
+    def __call__(self, image: np.ndarray) -> np.ndarray:
+
+        # add random offset
+        self.randomize(img_size=image.shape)
+
+        if self.tile_random_offset and self.random_offset is not None:
+            image = image[:, self.random_offset[0] :, self.random_offset[1] :]
+
+        # pad to full size, divisible by tile_size
+        if self.tile_pad_full:
+            c, h, w = image.shape
+            pad_h = (self.tile_size - h % self.tile_size) % self.tile_size
+            pad_w = (self.tile_size - w % self.tile_size) % self.tile_size
+            image = np.pad(
+                image,
+                [[0, 0], [pad_h // 2, pad_h - pad_h // 2], [pad_w // 2, pad_w - pad_w // 2]],
+                constant_values=self.tile_background_val,
+            )
+
+        # extact tiles (new way)
+        xstep, ystep = self.tile_step, self.tile_step
+        xsize, ysize = self.tile_size, self.tile_size
+        clen, xlen, ylen = image.shape
+        cstride, xstride, ystride = image.strides
+        llw = as_strided(
+            image,
+            shape=((xlen - xsize) // xstep + 1, (ylen - ysize) // ystep + 1, clen, xsize, ysize),
+            strides=(xstride * xstep, ystride * ystep, cstride, xstride, ystride),
+            writeable=False,
+        )
+        image = llw.reshape(-1, clen, xsize, ysize)
+
+        # if keep all patches
+        if self.tile_all:
+            # retain only patches with significant foreground content to speed up inference
+            # FYI, this returns a variable number of tiles, so the batch_size much be 1 (per gpu). Used during inference
+            thresh = 0.999 * 3 * self.tile_background_val * self.tile_size * self.tile_size
+            if self.tile_filter_mode == "min":
+                # default, keep non-background tiles (small values)
+                idxs = np.argwhere(image.sum(axis=(1, 2, 3)) < thresh)
+                image = image[idxs.reshape(-1)]
+            elif self.tile_filter_mode == "max":
+                idxs = np.argwhere(image.sum(axis=(1, 2, 3)) >= thresh)
+                image = image[idxs.reshape(-1)]
+
+        else:
+            if len(image) >= self.tile_count:
+
+                if self.tile_filter_mode == "min":
+                    # default, keep non-background tiles (smallest values)
+                    idxs = np.argsort(image.sum(axis=(1, 2, 3)))[: self.tile_count]
+                    image = image[idxs]
+                elif self.tile_filter_mode == "max":
+                    idxs = np.argsort(image.sum(axis=(1, 2, 3)))[-self.tile_count :]
+                    image = image[idxs]
+                elif len(image) > self.tile_count:
+                    # random subset (more appropriate for WSIs without distinct background)
+                    if self.random_idxs is not None:
+                        image = image[self.random_idxs]
+
+            else:
+                image = np.pad(
```

Comment:
I would suggest to use MONAI `Pad` transform:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/croppad/array.py#L72

Thanks.

Reply:
I don't see a reason for Pad here.  A simple numpy.pad is concise ,  and easy to read.